ModComponent/Mapper/ZipFileLoader.cs

```diff
@@ -84,17 +81,9 @@ private static void LoadZipFile(string zipFilePath)
 		}
 	}
 
-	private static Encoding GetEncoding(MemoryStream memoryStream)
-	{
-		using StreamReader reader = new StreamReader(memoryStream, true);
-		reader.Peek();
-		return reader.CurrentEncoding;
-	}
-
 	private static string ReadToString(MemoryStream memoryStream)
 	{
-		Encoding encoding = GetEncoding(memoryStream);
-		return encoding.GetString(memoryStream.ToArray());
+		return Encoding.UTF8.GetString(memoryStream.ToArray());
```

Comment:
I think this can directly use the stream without converting to an array.

Reply:
`Argument 1: cannot convert from 'System.IO.MemoryStream' to 'byte[]'`

Reply:
`return new StreamReader(memoryStream, Encoding.UTF8).ReadToEnd();`

Reply:
You may have to check the first few bytes for BOM and possibly use a cached `new Utf8Encoding(false)`.

Reply:
Implemented same BOM check as LocalizationUtils

Reply:
That was fine in LocalizationUtilities because the file would always be json, but not here. I want a more general purpose solution.
```cs
	private static string ReadToString(MemoryStream memoryStream)
	{
		ReadOnlySpan<byte> utf8BOM = stackalloc byte[] { 0xEF, 0xBB, 0xBF };
		ReadOnlySpan<byte> streamData = memoryStream.ToArray();
		ReadOnlySpan<byte> textData = streamData.StartsWith(utf8BOM) ? streamData.Slice(3) : streamData;
		return Encoding.UTF8.GetString(textData);
	}
```